src/generic/GenericTypes.jl

```diff
@@ -1073,7 +1078,10 @@ end
 
 # construct zero matrix
 function MatSpaceElem{T}(R::NCRing, r::Int, c::Int) where T <: NCRingElement
-   entries = fill(zero(R), r, c)
+   entries = Matrix{T}(undef, r, c)
+   for i in 1:r*c
+      entries[i] = zero(R)
```

Comment:
Note: this is actually a bug fix, so even if we reject this PR here, this part of it should be merged... The bug is that by aliasing some entries, some code won't work right if it uses functions like `add!` on matrix entries... I found out because this PR now uses this constructor, so the issue became apparent; in `master` it lurks hidden.

Reply:
Yes, the code in AA assumes that one cannot do `add!` on matrix entries per se. If we want to change this, this needs further discussion and maybe do it in a separate PR.

Reply:
So are you saying that code I found which failed when I aliased the zeros is wrong?

Otherwise, that link says that matrix elements may alias; but it does not prescribe whether e.g. `zero_matrix` may produce a matrix with aliased entries; or whether it perhaps *must* produce one with non-aliased entries?

For now I went with the latter because it fixed a bug, and it also preserves the current behavior, so is less likely to introduce regressions.

Reply:
Yes, it is probably wrong. But I would have to see it.

Reply:
Hm, maybe the same error as in https://github.com/Nemocas/AbstractAlgebra.jl/pull/955 :) There Claus made things also correct according to the documentation, but it exposed some other bugs.

I will have a look

Reply:
I wrote something about this proposed change in #1000